yt_dlp/extractor/chzzk.py

```diff
@@ -146,19 +146,39 @@ def _real_extract(self, url):
         video_meta = self._download_json(
             f'https://api.chzzk.naver.com/service/v3/videos/{video_id}', video_id,
             note='Downloading video info', errnote='Unable to download video info')['content']
-        formats, subtitles = self._extract_mpd_formats_and_subtitles(
-            f'https://apis.naver.com/neonplayer/vodplay/v1/playback/{video_meta["videoId"]}', video_id,
-            query={
-                'key': video_meta['inKey'],
-                'env': 'real',
-                'lc': 'en_US',
-                'cpl': 'en_US',
-            }, note='Downloading video playback', errnote='Unable to download video playback')
+        video_status = video_meta.get('vodStatus')
+        video_live_date = video_meta.get('liveOpenDate')
+        if video_live_date:
+            live_status = 'was_live'
+        else:
+            live_status = 'not_live'
 
+        if video_status == 'UPLOAD':
+            playback = self._parse_json(video_meta.get('liveRewindPlaybackJson'), video_id)
+            formats, subtitles = self._extract_m3u8_formats_and_subtitles(
+                traverse_obj(playback, ('media', 0, 'path')), video_id,
+                note='Downloading video playback', errnote='Unable to download video playback')
+        elif video_status == 'ABR_HLS':
+            formats, subtitles = self._extract_mpd_formats_and_subtitles(
+                f'https://apis.naver.com/neonplayer/vodplay/v1/playback/{video_meta.get("videoId")}', video_id,
+                query={
+                    'key': video_meta['inKey'],
+                    'env': 'real',
+                    'lc': 'en_US',
+                    'cpl': 'en_US',
+                }, note='Downloading video playback', errnote='Unable to download video playback')
+        else:
+            self.raise_no_formats(f'Unknown video status detected: "{video_status}"',
+                                  expected=True, video_id=video_id)
+            formats = []
+            subtitles = {}
+            if live_status == 'was_live':
+                live_status = 'post_live'
```

Comment:
just formatting/style nitpicks
```suggestion
            self.raise_no_formats(
                f'Unknown video status detected: "{video_status}"', expected=True, video_id=video_id)
            formats, subtitles = [], {}
            live_status = 'post_live' if live_status == 'was_live' else None
```